src/vivarium_gates_lsff/components/disease/iron_deficiency.py

```diff
@@ -190,9 +190,15 @@ def ppf(self, propensity: pd.Series) -> pd.Series:
         exposure_data = self.exposure_parameters(propensity.index)
         mean = exposure_data['mean']
         sd = exposure_data['sd']
-        exposure = (data_values.HEMOGLOBIN_DISTRIBUTION.WEIGHT_GAMMA * self._gamma_ppf(propensity, mean, sd)
-                    + data_values.HEMOGLOBIN_DISTRIBUTION.WEIGHT_GUMBEL * self._mirrored_gumbel_ppf(propensity, mean, sd))
-        return pd.Series(exposure, index=propensity.index, name='value')
+
+        ensemble_propensity = self.ensemble_propensity.get_draw(propensity.index)
```

Comment:
You don't want to do this every time step.  Get a propensity during simulant initialization and cache it.  Otherwise you'll have individuals jumping back and forth between anemia levels.  This will average out in the aggregate statistics, but produce very unrealistic individual trajectories.

Reply:
Whoops, wrong account :-)

Reply:
OK, PR updated. Compared to the last change we now have more simulants in the mild and moderate categories

  Single propensity    Per-step propensity
  anemia_none           26867.619438740585 : 27537.127994524297
  anemia_mild           3291.8877481177274 : 3084.4079397672826
  anemia_moderate   1019.6139630390143 : 558.466803559206
  anemia_severe       1.1882272416153319 : 0.30663928815879526